### PR TITLE
Plumb terratest configuration

### PR DESCRIPTION
**Pull request by dynamike**
Now that we've got some churn on this module, I'm adding tests. This adds a test to spin up SES on a temporary domain and configure email bounces to go to an S3 bucket. 

```
--- PASS: TestTerraformSESDomain (145.15s)
PASS
ok  	github.com/trussworks/terraform-aws-ses-domain/test	145.458
```

### CODE REVIEW

File: examples/simple/main.tf [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  infra_test_truss_coffee = "infra-test.truss.coffee"
+  temp_domain             = "${var.test_name}.${local.infra_test_truss_coffee}"
+}
+
+resource "aws_ses_receipt_rule_set" "main" {
+  rule_set_name = var.test_name
+}
+
+resource "aws_ses_active_receipt_rule_set" "main" {
+  rule_set_name = var.test_name
```
Comment (jsclarridge) [inline examples/simple/main.tf]:
```suggestion
  rule_set_name = aws_ses_receipt_rule_set.main.rule_set_name
```

File: examples/simple/main.tf [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+  depends_on = [aws_ses_receipt_rule_set.main]
```
Comment (jsclarridge) [inline examples/simple/main.tf]:
```suggestion
```

File: examples/simple/main.tf [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  name = local.temp_domain
+}
+
+resource "aws_route53_record" "temp_domain_ns_records" {
+  zone_id = data.aws_route53_zone.infra_test_truss_coffee.zone_id
+  name    = local.temp_domain
+  type    = "NS"
+  ttl     = "30"
+
+  records = aws_route53_zone.temp_domain.name_servers.*
```
Comment (jsclarridge) [inline examples/simple/main.tf]:
```suggestion
  records = aws_route53_zone.temp_domain.name_servers
```

File: examples/simple/main.tf [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+  from_addresses   = ["no-reply@${local.temp_domain}"]
+  mail_from_domain = "email.${local.temp_domain}"
+
+  dmarc_rua = "email@hurts.com"
+
+  receive_s3_bucket = module.ses_bucket.id
+  receive_s3_prefix = local.ses_bucket_prefix
+
+  ses_rule_set = var.test_name
```
Comment (jsclarridge) [inline examples/simple/main.tf]:
```suggestion
  ses_rule_set = aws_ses_receipt_rule_set.main.rule_set_name
```

**Comment by dynamike** [comment]
applied and tested

File: examples/simple/main.tf [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+  temp_domain             = "${var.test_name}.${local.infra_test_truss_coffee}"
+}
+
+resource "aws_ses_receipt_rule_set" "main" {
+  rule_set_name = var.test_name
+}
+
+resource "aws_ses_active_receipt_rule_set" "main" {
+  rule_set_name = aws_ses_receipt_rule_set.main.rule_set_name
+
```
Comment (jsclarridge) [inline examples/simple/main.tf]:
```suggestion
```

Comment (jsclarridge) [reply examples/simple/main.tf]: I know this is nit-picky, but maybe delete that blank line too...

**Pull request merged**